assertj-core/src/main/java/org/assertj/core/api/AbstractBigDecimalAssert.java

```diff
@@ -523,4 +543,30 @@ public AbstractBigDecimalScaleAssert<SELF> scale() {
     requireNonNull(actual, "Can not perform assertions on the scale of a null BigDecimal");
     return new BigDecimalScaleAssert(myself);
   }
+
+  /**
+   * Returns an {@code Assert} object that allows performing assertions on the precision of the {@link BigDecimal} under test.
+   * <p>
+   * Once this method is called, the object under test is no longer the {@link BigDecimal} but its precision.
+   * To perform assertions on the {@link BigDecimal}, call {@link AbstractBigDecimalPrecisionAssert#returnToBigDecimal()}.
+   * <p>
+   * Example:
+   * <pre><code class='java'> BigDecimal bgDecimal = new BigDecimal(&quot;9.3231&quot;);
+   *
+   * // assertions succeed
+   * assertThat(bgDecimal).precision().isGreaterThan(1)
+   *                                  .isLessThan(6)
+   *                      .returnToBigDecimal().isPositive();
+   *
+   * // assertions fails
+   * assertThat(bgDecimal).precision().isBetween(6, 8);</code></pre>
+   *
+   * @return AbstractBigDecimalPrecisionAssert built with the {@code BigDecimal}'s precision.
+   * @throws NullPointerException if the given {@code BigDecimal} is {@code null}.
+   */
+  @SuppressWarnings({ "unchecked", "rawtypes" })
+  public AbstractBigDecimalPrecisionAssert<SELF> precision() {
+    requireNonNull(actual, "Can not perform assertions on the precision of a null BigDecimal");
+    return new BigDecimalPrecisionAssert(myself);
```

Comment:
We briefly discussed this approach with the team.

We should add a bit of context to the error messages by showing the root object we navigated from.

For example, assuming:

```
assertThat(myBigDecimal).scale().isGreaterThan(5L)
```

If the assertion fails, it will show something like

```
expecting 2 to be greater than 5 but was not
```

where the fact that a `BigDecimal` precision was checked is forgotten.

Instead, the error message should have a better description and be more like:

```
[checking 2.123 precision] expecting 2 to be greater than 5 but was not
```

Unfortunately, we don't do this consistently in AssertJ so I didn't manage to find a good example quickly. I'll try to have another look in the next few days.